elements/lisk-transactions/src/sign.ts

```diff
@@ -145,6 +145,11 @@ export const signTransaction = (
 		throw new Error('Private key must be 64 bytes');
 	}
 
+	if (transactionObject.params && Object.keys(transactionObject.params).length === 0) {
+		// eslint-disable-next-line no-param-reassign
+		transactionObject.params = null;
```

Comment:
i think mutating object from user is not so good.
it would be nicer if we check the paramsSchema in the `getSigningBytes` function

Reply:
Would it be ok to mutate the params object in `getSigningBytes`? A simple check without mutating the value of the empty params object will persist the bug.

Reply:
I think we can simply check the `paramsSchema` in [here](https://github.com/LiskHQ/lisk-sdk/pull/8029/files/03e62301f7dfeaa2606114ef10867aee048e8ed4#diff-2ca6db1bdc8b33bc7e77c47bb4f272240031d265f2a028ad2021790117aa377bR57) additionally like below?
```
(typeof transactionObject.params !== 'object' || transactionObject.params === null || !paramsSchema)
```

Reply:
Would you kindly check if the link works, seems to be linking to a blank commit.

Reply:
ah sorry, i think link not changed in the PR doesnt work. it's L62 on `elements/lisk-transactions/src/sign.ts`

Reply:
> `(typeof transactionObject.params !== 'object' || transactionObject.params === null || !paramsSchema)`

...still throws as `paramsSchema` can be undefined and won't cause an issue. The issue is `transactionObject.params` being an empty object.

Reply:
if it goes in the condition,
```
const transactionBytes = codec.encode(baseTransactionSchema, {
			...transactionObject,
			params: Buffer.alloc(0),
			signatures: [],
		});
```
I think params is not used to get the bytes, so why would it be a problem?

Reply:
Throws a `TypeError: Cannot read properties of undefined (reading '$id')`